packages/callback-handler-svc/src/server/index.ts

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable camelcase */
+import express from 'express'
+import http from 'http'
+import Config from '../shared/config'
+import Logger from '@mojaloop/central-services-logger'
+import Metrics from '@mojaloop/central-services-metrics'
+import { logger } from '~/shared/logger'
+
+type TracestateMap = {
+  tx_end2end_start_ts: number | undefined;
+  tx_callback_start_ts: number | undefined;
+}
+const app = express()
+let appInstance: http.Server
+
+function getTraceStateMap (headers: any): TracestateMap {
+  const tracestate: string = headers.tracestate
+  if (tracestate === undefined) {
+    return {
+      tx_end2end_start_ts: undefined,
+      tx_callback_start_ts: undefined
+    }
+  }
+  let tracestates = {}
+  tracestate
+    .split(',')
+    .map(ts => ts.split('='))
+    .map(([k, v]) => {
+      return { [k]: Number(v) }
+    })
+    .forEach(ts => {
+      tracestates = { ...tracestates, ...ts }
+    })
+  return tracestates as TracestateMap
+}
+
+async function run (): Promise<void> {
+  logger.info(Config)
+  if (!Config.INSTRUMENTATION.METRICS.DISABLED) {
+    Metrics.setup(Config.INSTRUMENTATION.METRICS.config)
+  }
+
+  app.use(express.json())
+  app.get('/health', (_req, res) => {
+    res.json({
+      status: 'OK'
+    })
+  })
+  app.get('/metrics', async (_req, res) => {
+    res.status(200)
+    res.send(await Metrics.getMetricsForPrometheus())
+  })
+  app.get(['/:operation', '/:operation/*'], async (req, res) => {
+    const currentTime = Date.now()
+    const path = req.path
+    const isErrorOperation = path.endsWith('error')
+    const operation = req.params.operation
+    const operationE2e = `${operation}_end2end`
+    const operationRequest = `${operation}_request`
+    const operationResponse = `${operation}_response`
+    const tracestate = getTraceStateMap(req.headers)
+
+    if (tracestate.tx_end2end_start_ts === undefined || tracestate.tx_callback_start_ts === undefined) {
+      return res.status(400).send('tx_end2end_start_ts or tx_callback_start_ts key/values not found in tracestate')
+    }
+
+    const e2eDelta = currentTime - tracestate.tx_end2end_start_ts
+    const requestDelta = tracestate.tx_callback_start_ts - tracestate.tx_end2end_start_ts
+    const responseDelta = currentTime - tracestate.tx_callback_start_ts
+
+    const performanceHistogram = Metrics.getHistogram(
+      'cb_perf',
+      'Metrics for callbacks',
+      ['success', 'path', 'operation']
+    )
+
+    performanceHistogram.observe({
+      success: (!isErrorOperation).toString(),
+      path,
+      operation: operationE2e
+    }, e2eDelta / 1000)
+    performanceHistogram.observe({
+      success: (!isErrorOperation).toString(),
+      path,
+      operation: operationRequest
+    }, requestDelta / 1000)
+    performanceHistogram.observe({
+      success: (!isErrorOperation).toString(),
+      path,
+      operation: operationResponse
+    }, responseDelta / 1000)
+
+    return res.json({
+      tracestate,
```

Comment:
Can we include the `traceparent` header at a minimum here:

```suggestion
      traceparent,
      tracestate,
```

Alternatively, we could include the trace details separately as in `trace-id`, `parent-id`, etc.

Reply:
done. added traceparent.